distributed-testing/src/main/groovy/net/corda/testing/ImageBuilding.java

```diff
@@ -0,0 +1,152 @@
+package net.corda.testing;
+
+import com.bmuschko.gradle.docker.DockerRegistryCredentials;
+import com.bmuschko.gradle.docker.tasks.container.*;
+import com.bmuschko.gradle.docker.tasks.image.*;
+import org.gradle.api.GradleException;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * this plugin is responsible for setting up all the required docker image building tasks required for producing and pushing an
+ * image of the current build output to a remote container registry
+ */
+public class ImageBuilding implements Plugin<Project> {
+
+    public static final String registryName = "stefanotestingcr.azurecr.io/testing";
```

Comment:
Can we "De-Stefano-ise" this please?

Reply:
In the future yes but not as part of this